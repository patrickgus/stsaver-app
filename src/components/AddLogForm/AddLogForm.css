.AddLogForm__form-section {
  margin-bottom: 15px;
}

.AddLogForm__form-section label {
  margin-right: 5px;
}

.times input {
  width: 80px;
}

.AddLogForm__buttons {
  display: flex;
  justify-content: space-evenly;
}

#breaks {
  width: 50px;
}
