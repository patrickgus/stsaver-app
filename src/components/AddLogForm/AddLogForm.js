import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import "./AddLogForm.css";

class AddLogForm extends Component {
  handleSubmit = e => {
    e.preventDefault();
    this.props.history.push("/activity");
  };

  handleCancel = e => {
    e.preventDefault();
    this.props.history.push("/activity");
  };

  render() {
    return (
      <form className="AddLogForm" onSubmit={this.handleSubmit}>
        <div className="AddLogForm__form-section">
          <label htmlFor="date">Date</label>
          <input type="date" name="date" id="date" required />
        </div>
        <div className="AddLogForm__form-section times">
          <label htmlFor="start-time">Start time</label>
          <input type="time" name="start-time" id="start-time" required />
          <label htmlFor="end-time">End time</label>
          <input type="time" name="end-time" id="end-time" required />
        </div>
        <div className="AddLogForm__form-section">
          <label htmlFor="media-type">Media type</label>
          <select name="media-type" id="media-type" required>
            <option value="">--Choose one--</option>
            <option value="computer">Computer</option>
            <option value="phone">Phone</option>
            <option value="tablet">Tablet</option>
            <option value="television">Television</option>
          </select>
        </div>
        <div className="AddLogForm__form-section">
          <label htmlFor="breaks">Number of breaks</label>
          <input type="number" name="breaks" id="breaks" required />
        </div>
        <button type="submit">Submit</button>
        <button onClick={this.handleCancel}>Cancel</button>
      </form>
    );
  }
}

export default withRouter(AddLogForm);
