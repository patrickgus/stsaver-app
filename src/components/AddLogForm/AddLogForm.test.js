import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import AddLogForm from "./AddLogForm";

describe("AddLogForm component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <AddLogForm />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AddLogForm />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
