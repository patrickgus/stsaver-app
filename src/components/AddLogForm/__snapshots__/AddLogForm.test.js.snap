// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddLogForm component renders the UI as expected 1`] = `
<form
  className="AddLogForm"
  onSubmit={[Function]}
>
  <div
    className="AddLogForm__form-section"
  >
    <label
      htmlFor="date"
    >
      Date
    </label>
    <input
      id="date"
      name="date"
      required={true}
      type="date"
    />
  </div>
  <div
    className="AddLogForm__form-section times"
  >
    <label
      htmlFor="start-time"
    >
      Start time
    </label>
    <input
      id="start-time"
      name="start-time"
      required={true}
      type="time"
    />
    <label
      htmlFor="end-time"
    >
      End time
    </label>
    <input
      id="end-time"
      name="end-time"
      required={true}
      type="time"
    />
  </div>
  <div
    className="AddLogForm__form-section"
  >
    <label
      htmlFor="media-type"
    >
      Media type
    </label>
    <select
      id="media-type"
      name="media-type"
      required={true}
    >
      <option
        value=""
      >
        --Choose one--
      </option>
      <option
        value="computer"
      >
        Computer
      </option>
      <option
        value="phone"
      >
        Phone
      </option>
      <option
        value="tablet"
      >
        Tablet
      </option>
      <option
        value="television"
      >
        Television
      </option>
    </select>
  </div>
  <div
    className="AddLogForm__form-section"
  >
    <label
      htmlFor="breaks"
    >
      Number of breaks
    </label>
    <input
      id="breaks"
      name="breaks"
      required={true}
      type="number"
    />
  </div>
  <div
    className="AddLogForm__buttons"
  >
    <button
      type="submit"
    >
      Submit
    </button>
    <button
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</form>
`;
